docker/cache.go

```diff
@@ -7,8 +7,9 @@ import (
 
 // bicTransportScope returns a BICTransportScope appropriate for ref.
 func bicTransportScope(ref dockerReference) types.BICTransportScope {
-	// Blobs can be reused across the whole registry.
-	return types.BICTransportScope{Opaque: reference.Domain(ref.ref)}
+	// Blobs can be reused across multiple registries
+	// therefore bucket all blobs in same scope
+	return types.BICTransportScope{Opaque: "all"}
```

Comment:
This looses most of the nuance/intelligence. We do want `CandidateLocations2` to primarily try to use blobs that are known to exist on that registry, not just recently-used digests — _only if_ we have no candidates at that location, blindly try the compressed version (and only a few such digests — compare `prioritize.replacementAttempts`).

Reply:
One issue is that we will perform lookup twice one iteration for `known` blobs via `CandidateLocations2` and another if we create a new function to perform lookup on all available `blobs`. So there is a overlap of while searching for `blobs` of course we could minimize that by some checks.

So i agree we can split the matching of `all` away from `CandidateLocations2` so original logic remains intact. Let try something around this.

Reply:
`CandidateLocations2` is an internal interface, we can reshape it any way we like.

Reply:
@mtrmac just re-visiting this since BZ showed interest in this RFE again, do you like the idea if we modify `CandidateLocations2` to return `all` blobs but give these newly added blobs will have least priority so maybe some modifications in `DestructivelyPrioritizeReplacementCandidates`.

Reply:
- I agree with the lowest priority part for “location-unknown” candidates. (We should also limit their number.)
- `Opaque` does mean `Opaque` to the generic code: that field is exclusively up to the transport to define and the generic cache code has no business checking for specific string values. In the returned `BICReplacementCandidate2`, introduce a new way to represent “location unknown” values (an extra bool or turn `Location` into a pointer).

https://github.com/containers/image/pull/550#pullrequestreview-579816214 is an old description how I imagine this could work — now I don’t think we need a separate `CandidateDigests`, just let `CandidateLocations2` include the location-unknown results. But that’s just a guess and the structure might end up different based on the final code.

Reply:
@mtrmac Got it i have a patch ready, do you prefer if i open a new PR cause i think it might have different review comments altogether and close this PR ? But i can also force push on this PR if that is preferred.